swap/int256/int256.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package int256
+
+import (
+	"fmt"
+	"io"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// Int256 represents an signed integer of 256 bits
+type Int256 struct {
+	value big.Int
+}
+
+var minInt256 = new(big.Int).Mul(big.NewInt(-1), new(big.Int).Exp(big.NewInt(2), big.NewInt(255), nil)) // -(2^255)
+var maxInt256 = new(big.Int).Sub(new(big.Int).Exp(big.NewInt(2), big.NewInt(255), nil), big.NewInt(1))  // 2^255 - 1
+
+// NewInt256 creates a Int256 struct with an initial underlying value of zero
+func NewInt256() *Int256 {
+	u := new(Int256)
+	u.value = *new(big.Int).Set(big.NewInt(0))
+	return u
+}
+
+// Value returns the underlying private value for a Int256 struct
+func (u *Int256) Value() big.Int {
+	return u.value
+}
+
+// Int256From creates a Int256 struct based on the given int64 param
+// any int64 is valid as a Int256
+func Int256From(base int64) *Int256 {
+	u := NewInt256()
+	u.value = *new(big.Int).SetInt64(base)
+	return u
+}
+
+// Set assigns the underlying value of the given Int256 param to u, and returns the modified receiver struct
+// returns an error when the result falls outside of the unsigned 256-bit integer range
+func (u *Int256) Set(value big.Int) (*Int256, error) {
+	if value.Cmp(maxInt256) == 1 {
```

Comment:
i vote to get rid of this and have this functionality in the constructor. it will make a lot of things much cleaner in this PR, plus it does not allow the possibility of overriding a value on an existing value instance

Reply:
done